src/librustc_typeck/mem_categorization.rs

```diff
@@ -516,8 +526,17 @@ impl<'a, 'tcx> MemCategorizationContext<'a, 'tcx> {
         ty: Ty<'tcx>,
     ) -> PlaceWithHirId<'tcx> {
         let mut projections = base_place.place.projections;
-        projections.push(Projection { kind: ProjectionKind::Other });
-        let ret = PlaceWithHirId::new(node.hir_id(), ty, base_place.place.base, projections);
+        projections.push(Projection {
+            kind: ProjectionKind::Other,
+            before_ty: base_place.place.base_ty,
```

Comment:
Should this not be `base_place.place.ty()`?

Reply:
Done